proposals/devcontainer-collections.md

```diff
@@ -0,0 +1,110 @@
+# Proposal
+
+The proposal is to arrive to an schema where different communities and organizations can create "collections" of features/definitions that they manage, support and present to users/
+
+This collection data should have enough information so that different clients can search them and present them to users.
+
+For this we propose the existence of two different files:
+- Collection file: (devcontainer-collection.json) Groups of features/definitions that will be handled as the same origin (repository/publisher). Contains the metadata pertinent for each one, checksum and other information.
+- Feature/Definition: (devcontainer-feature.json/devcontainer-definition.json) Information on a particular feature/definition required to add it to a collection. This file is optional and exists to support scenarios where one repository represents a collection and others represent each feature.
```

Comment:
I think we'll want to make a decision here, either:

1. Create one top-level  "devcontainer-collection.json` file that has  `features: [ ]` and `definitions: []` arrays storing the metadata.
2. Each individual feature and definition has their own "snippet" of metadata, that the GitHub action will then stitch together

Reply:
Got it.  For now i've placed all the definitions into the a single file in vscode-dev-container, but could easily stitch that together with the existing GitHub Action

https://github.com/microsoft/vscode-dev-containers/blob/main/devcontainer-collection.json